### ISSUE TITLE
[$250] Hybrid App Android : Showing a Fix banner when there is nothing to fix

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#49740]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.40-0
**Reproducible in staging?:** Y
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @jliexpensify 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1727237125506689

## Action Performed:

1. SmartScan receipt with a field that falls outside the Workspace rules (e.g. over $10 USD on Coffee Workspace)
2. Confirm that it gets added to current unsubmitted Report under the Workspace
3. Click on expense and change field (e.g. edit expense amount to reflect $10 USD)
4. Observe the Fix banner on Report
5. Exit app and open again
6. Observe the Fix banner on Report

## Expected Result:
Fix Banner replaced with the status of the report or the next step displayed
## Actual Result:
Fix banner displayed even after applying the fix
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/e89472f4-f245-495c-a1ef-c13f8fad5364

https://github.com/user-attachments/assets/2416785e-609f-4be5-93ab-5125ccc653df

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021839400983690827454</li>
        <li>Upwork Job ID: 1839400983690827454</li>
        <li>Last Price Increase: 2024-09-26</li>
    </ul>
</details>

**Comment by JKobrynski** [comment]
Hi, I'm Julian from Callstack - expert agency - and I would like to work on this issue.

**Comment by sakluger** [comment]
Thanks @JKobrynski!

**Comment by JKobrynski** [comment]
It actually happens on all platforms:

Mobile web:


https://github.com/user-attachments/assets/2406b5e1-33e5-423e-a21a-f0456e05c996

iOS:


https://github.com/user-attachments/assets/5e1b7797-5a30-4369-be2c-b347abe212a6

**Comment by JKobrynski** [comment]
I think the reason why this is happening might be that the `UpdateMoneyRequestAmountAndCurrency` API command doesn't update `reportNextStep_<id>` Onyx entry. After the amount is changed to a value below the limit, the Onyx entry still stores the message that was shown for an invalid amount. What do you think? Could it be done on the backend?

**Comment by JKobrynski** [comment]
@sakluger have you had the chance to take a look at my question above? 🙏

**Comment by sakluger** [comment]
I'll ask our internal engineers if they can help.

**Comment by sakluger** [comment]
No updates.

**Comment by sakluger** [comment]
From Tom:

> This is a violations/report next steps bug, features of which are being built in [#expense](https://expensify.slack.com/archives/C06ML6X0W9L). Best to move it to the wave-control project

**Comment by sakluger** [comment]
Reposted in the `#expense` Slack channel: https://expensify.slack.com/archives/C06ML6X0W9L/p1728514367110159

**Comment by sakluger** [comment]
@mountiny said he may be able to look into this on.

**Comment by sakluger** [comment]
Bumped the Slack thread.

**Comment by mountiny** [comment]
The PR was on hold and now its in review

**Comment by sakluger** [comment]
The PR is merged and deployed to staging, but QA is saying that it's not fully fixed: https://github.com/Expensify/Web-Expensify/pull/43937#issuecomment-2429364447.

**Comment by fabioh8010** [comment]
Hey 👋 Just FYI that @JKobrynski is on sick leave since yesterday and is expected to return next week.

**Comment by JKobrynski** [comment]
Hey! FYI I'm back from the sick leave, I'll continue working on this issue

**Comment by JKobrynski** [comment]
> Seems like for this specific violation flow, there is a bug in the next steps logic as the cache for the expense is not recomputed, I will need to follow up on this one

Do you think there is anything that can be done on the frontend or is this backend only?

CC: @mountiny

**Comment by mountiny** [comment]
Havent got around yet

**Comment by mountiny** [comment]
Created a follow up that should hopefully fix this issue with cache

**Comment by mountiny** [comment]
Same,

**Comment by mountiny** [comment]
@dangrous is leading the way on fixing this in the backend by making sure the violations are always recomputed

**Comment by dangrous** [comment]
Fix PR is merged, not yet deployed. Hopefully we can retest and close this soon!

**Comment by dangrous** [comment]
Fix is on staging, can we retest this and close?

**Comment by sakluger** [comment]
I asked for a retest: https://expensify.slack.com/archives/C9YU7BX5M/p1734621671015609

**Comment by dangrous** [comment]
An updated version of the fix for this is now in review - the original fix was unfortunately reverted so we don't need the retest yet... Given the holidays though the timing might work out either way?

**Comment by dangrous** [comment]
Fix in review, I need to respond to comments now that I'm back from holiday break. Soon!

**Comment by dangrous** [comment]
The fix is ready, but we're debating a bit some performance tradeoffs that are seemingly required for the fix. Ideally we'll have resolution soon though

**Comment by dangrous** [comment]
Getting there! Both PRs that should fix this have been moved forward (discussion complete) and we should be nearly there.

**Comment by dangrous** [comment]
Waiting on deploy still

**Comment by dangrous** [comment]
Okay! I think this has been deployed and is ready for re-testing. Should we ping QA, or what's the next step?

Thanks!

**Comment by dangrous** [comment]
adding test label

**Comment by dangrous** [comment]
do we need to do anything else here to test/close?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by dangrous** [comment]
Great! I think I'll go ahead and close this. @rushatgabhane I see you're assigned here but it's been a long time so I don't remember who did what. If you need additional payment please comment here and I can reopen!
